packages/edit-site/src/components/template-part-modal/index.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import TemplatePartRenameModal from './rename';
+
+export const TEMPLATE_PART_MODALS = {
+	rename: 'edit-site/template-part-rename',
+	duplicate: 'edit-site/template-part-duplicate',
+};
+
+export default function TemplatePartModal() {
+	// Duplication command and modal is in separate follow-up.
+	return (
+		<>
+			<TemplatePartRenameModal />
```

Comment:
Will it still make sense to later consolidate the template-part and patterns rename modal logic into one component?  Just flagging the WIP PR over in: https://github.com/WordPress/gutenberg/pull/54271/files#diff-b19d86a4e904ec9e800ba7bc553c9687305444b9164e3eabba56abd57cc825d7

And would the long-term play be to remove the distinction completely between template parts and patterns?

Regardless, I like the way this PR abstracts the modals - will make acting on any consolidation decisions a little easier having distinct logical compartments 👍🏻

Reply:
> Will it still make sense to later consolidate the template-part and patterns rename modal logic into one component?

This is the main question I'm trying to settle on an answer for.

I appreciate that having only one component makes some things easier but it gets more abstract and murky. It also increases the potential that it gets changed for one post type without really understanding the impact that might have on the others. This is a bit of concern I hold for when it comes time to try and unify template parts and patterns.

For the record, I'm not opposed to merging all the underlying rename modal components and then using that for all the rename commands, the rename menu item etc. I'd like to understand where all the overlaps are first though.

As for the `rename-menu-item` file linked, that retrieves the entity record so could still leverage the new dedicated rename modals. If one was added for templates as well then that `RenameMenuItem` component becomes a simple wrapper. 


> And would the long-term play be to remove the distinction completely between template parts and patterns?

That's my understanding of the goals for template parts and patterns.

> Regardless, I like the way this PR abstracts the modals

That's the general feedback I've received when discussing how all this has been organised. That feedback doesn't cover a broad cross-section of the community though so it could change.

Do you think we are best served by finishing off these features and then working on further refactors as a code quality follow-up?

Reply:
Thanks for the explanation. 

> It also increases the potential that it gets changed for one post type without really understanding the impact that might have on the others. 
> I'd like to understand where all the overlaps are first though.

Yes, makes sense to see where the moving parts are first and get (and keep) things working, before shuffling things around

> Do you think we are best served by finishing off these features and then working on further refactors as a code quality follow-up?

100% Good call. It'd be good to bring consistent features to the actions menu but it doesn't have to be through consolidation for now. 

Organizing components like this is great step forward, thank you!

Reply:
Agree 100% on the consistent features 👍